back/engines/commercial/public_api/app/serializers/public_api/v2/reaction_serializer.rb

```diff
@@ -11,4 +11,28 @@ class PublicApi::V2::ReactionSerializer < PublicApi::V2::BaseSerializer
   )
 
   attribute(:reactable_type) { classname_to_type(object.reactable_type) }
+
+  attribute(:idea_id) do
+    if object.reactable_type == 'Idea'
+      object.reactable.id
+    elsif object.reactable_type == 'Comment'
+      object.reactable.post_type == 'Idea' ? object.reactable.post_id : nil
+    end
+  end
+
+  attribute(:project_id) do
+    if object.reactable_type == 'Idea'
+      object.reactable&.project_id
+    elsif object.reactable_type == 'Comment'
+      object.reactable&.post_type == 'Idea' ? object.reactable&.post&.project_id : nil
+    end
+  end
+
+  attribute(:initiative_id) do
+    if object.reactable_type == 'Initiative'
+      object.reactable&.id
+    elsif object.reactable_type == 'Comment'
+      object.reactable&.post_type == 'Initiative' ? object.reactable&.post_id : nil
+    end
+  end
```

Comment:
Out of consistency (if I remember correctly), I would go with something like that:
```suggestion
  attribute(:post_type) do
    case object.reactable_type
    when 'Idea', 'Initiative'
      object.reactable_type
    when 'Comment'
      object.reactable.post_type
    end
  end

  attribute(:post_id) do
    case object.reactable_type
    when 'Idea', 'Initiative'
      object.reactable.id
    when 'Comment'
      object.reactable.post_id
    end
  end

  attribute(:project_id) do
    case object.reactable_type
    when 'Idea'
      object.reactable.project_id
    when 'Comment'
      object.reactable.post.project_id if object.reactable.post_type == 'Idea'
    end
  end
```

But again, I'm not sure to understand why we need to add this redundant information.

Reply:
We need this information to just give extra meta information in powerbi that would be very convoluted to get it within PowerBI - but it's a good point about just using `post_id` instead of breaking out idea_id and initiative_id - made me realise that we can jsut rename those attributes when we create the tables in PBI

Reply:
OK, makes sense if they are difficult to get from the `reactable_id` in PowerBI 👍